packages/shared/src/authz/domain/authErrors.ts

```diff
@@ -1,17 +1,42 @@
-export type AuthError<ErrorCode extends string = string> = {
-  code: ErrorCode
-  message: string
+export type AuthError<ErrorCode extends string = string, Payload = undefined> = {
+  readonly code: ErrorCode
+  readonly message: string
+  readonly payload: Payload
 }
 
-export const defineAuthError = <ErrorCode extends string>(params: {
+export const defineAuthError = <
```

Comment:
Had to rework errors cause I arrived at a use case that needed them to be customizable w/ custom payloads (like VError.info().

This is the best DX I could achieve with these requirements in mind:
* the error types are actual dynamic classes so you can instantiate them like ones (e.g. new Error()
* you're forced to define the code and its accessible both on the instance and also the class itself
* you can attach a dynamic payload, and you don't have to even specify it if there is none (type = undefined)

only ugly thing about this is having to do `InstanceType<typeof FooError>` wherever you want to reference the error type